tests/integration/test_item.py

```diff
@@ -0,0 +1,110 @@
+# Copyright 2021-2022 NVIDIA Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import random
+
+import numpy as np
+import pytest
+from legate.core import LEGATE_MAX_DIM
+
+import cunumeric as num
+
+
+@pytest.mark.xfail
+def test_no_item():
+    shape = (3, 3)
+    arr_num = num.random.randint(0, 3, size=shape)
+    arr_np = np.array(arr_num)
+
+    expected_exc = KeyError
+    with pytest.raises(expected_exc):
+        arr_np.item()
+        # Numpy raises KeyError: 'invalid key'
+    with pytest.raises(expected_exc):
+        arr_num.item()
+        # cuNumeric raises ValueError: can only convert an array
+        # of size 1 to a Python scalar
+
+
+@pytest.mark.xfail
+def test_out_of_bound():
+    shape = (3, 3)
+    arr_num = num.random.randint(0, 3, size=shape)
+    arr_np = np.array(arr_num)
+    expected_exc = IndexError
+    with pytest.raises(expected_exc):
+        arr_np.item(10)
+        # Numpy raises IndexError: index 10 is out of bounds for size 9
+    with pytest.raises(expected_exc):
+        arr_num.item(10)
+        # cuNumeric returns some value
+
+
+@pytest.mark.xfail
+def test_out_of_index():
+    shape = (3, 3)
+    arr_num = num.random.randint(0, 3, size=shape)
+    arr_np = np.array(arr_num)
+    expected_exc = IndexError
+    with pytest.raises(expected_exc):
+        arr_np.item(2, 4)
+        # Numpy raises IndexError: index 4 is out of bounds
+        # for axis 1 with size 3
+    with pytest.raises(expected_exc):
+        arr_num.item(2, 4)
+        # cuNumeric raises ValueError: Out-of-bounds projection on dimension 0
+        # with index 4 for a store of shape Shape((3,))
+
+
+def test_empty_no_item():
+    shape = ()
+    arr_num = num.random.randint(0, 3, size=shape)
+    arr_np = np.array(arr_num)
+
+    res_np = arr_np.item()
+    res_num = arr_num.item()
+    assert np.array_equal(res_np, res_num)
+
+
+def generate_item(ndim):
+    index_list = []
+    max_index = pow(4, ndim) - 1
+    random_index = random.randint(0, max_index)
+    index_list.append(random_index)
+    index_list.append(max_index)
+    random_tuple = []
+    for i in range(0, ndim):
+        random_x = random.randint(0, 3)
+        random_tuple.append(random_x)
+    index_list.append(tuple(random_tuple))
+    return index_list
```

Comment:
Is this the exact same as the version in the `test_itemset` module? Maybe factor one version into some helper or util module.